### PR TITLE
Shifts a lot of obj vars and code to flags

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
adds resistance_flags and obj_flags

## Why It's Good For The Game
moves us towards better code and some other refactors

### CODE REVIEW

**Comment by Alterist** [comment]
good shit

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -169,7 +168,7 @@
 
 
 /obj/machinery/door/ex_act(severity)
-	if(unacidable) return
+	if(CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE)) return
```
Comment (DominikPanic) [inline code/game/machinery/doors/door.dm]: Same line control statement.

File: code/game/objects/objs.dm [hunk 2]
```diff
@@ -42,42 +47,45 @@
...
-		for(var/mob/M in nearby)
-			if ((M.client && M.interactee == src))
+	if(!CHECK_BITFIELD(obj_flags, IN_USE))
+		return
+	var/is_in_use = 0
+	var/list/nearby = viewers(1, src)
+	for(var/mob/M in nearby)
+		if ((M.client && M.interactee == src))
+			is_in_use = 1
+			src.attack_hand(M)
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
			attack_hand(M)
```

File: code/game/objects/objs.dm [hunk 3]
```diff
@@ -42,42 +47,45 @@
...
+		if (!(usr in nearby))
+			if (usr.client && usr.interactee==src) // && M.interactee == src is omitted because if we triggered this by using the dialog, it doesn't matter if our machine changed in between triggering it and this - the dialog is probably still supposed to refresh.
 				is_in_use = 1
-				src.attack_hand(M)
-		if (isAI(usr) || iscyborg(usr))
-			if (!(usr in nearby))
-				if (usr.client && usr.interactee==src) // && M.interactee == src is omitted because if we triggered this by using the dialog, it doesn't matter if our machine changed in between triggering it and this - the dialog is probably still supposed to refresh.
-					is_in_use = 1
-					src.attack_ai(usr)
+				src.attack_ai(usr)
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
				attack_ai(usr)
```

File: code/game/objects/objs.dm [hunk 4]
```diff
@@ -42,42 +47,45 @@
...
-					if(usr.client && usr.interactee==src)
-						is_in_use = 1
-						src.attack_hand(usr)
-		in_use = is_in_use
+	if (ishuman(usr))
+		if(istype(usr.l_hand, /obj/item/tk_grab) || istype(usr.r_hand, /obj/item/tk_grab/))
+			if(!(usr in nearby))
+				if(usr.client && usr.interactee==src)
+					is_in_use = 1
+					src.attack_hand(usr)
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
					attack_hand(usr)
```

File: code/game/objects/objs.dm [hunk 5]
```diff
@@ -42,42 +47,45 @@
...
-				if(!(usr in nearby))
-					if(usr.client && usr.interactee==src)
-						is_in_use = 1
-						src.attack_hand(usr)
-		in_use = is_in_use
+	if (ishuman(usr))
+		if(istype(usr.l_hand, /obj/item/tk_grab) || istype(usr.r_hand, /obj/item/tk_grab/))
+			if(!(usr in nearby))
+				if(usr.client && usr.interactee==src)
+					is_in_use = 1
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
					is_in_use = TRUE
```

File: code/game/objects/objs.dm [hunk 6]
```diff
@@ -42,42 +47,45 @@
...
-		var/list/nearby = viewers(1, src)
-		for(var/mob/M in nearby)
-			if ((M.client && M.interactee == src))
+	if(!CHECK_BITFIELD(obj_flags, IN_USE))
+		return
+	var/is_in_use = 0
+	var/list/nearby = viewers(1, src)
+	for(var/mob/M in nearby)
+		if ((M.client && M.interactee == src))
+			is_in_use = 1
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
			is_in_use = TRUE
```

File: code/game/objects/objs.dm [hunk 7]
```diff
@@ -42,42 +47,45 @@
...
 
 /obj/proc/updateUsrDialog()
-	if(in_use)
-		var/is_in_use = 0
-		var/list/nearby = viewers(1, src)
-		for(var/mob/M in nearby)
-			if ((M.client && M.interactee == src))
+	if(!CHECK_BITFIELD(obj_flags, IN_USE))
+		return
+	var/is_in_use = 0
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
	var/is_in_use = FALSE
```

File: code/game/objects/objs.dm [hunk 8]
```diff
@@ -42,42 +47,45 @@
...
-				is_in_use = 1
-				src.interact(M)
-		var/ai_in_use = AutoUpdateAI(src)
-
-		if(!ai_in_use && !is_in_use)
-			in_use = 0
+	if(!CHECK_BITFIELD(obj_flags, IN_USE))
+		return
+	var/list/nearby = viewers(1, src)
+	var/is_in_use = 0
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
	var/is_in_use = FALSE
```

File: code/game/objects/objs.dm [hunk 9]
```diff
@@ -42,42 +47,45 @@
...
+	for(var/mob/M in nearby)
+		if ((M.client && M.interactee == src))
+			is_in_use = 1
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
			is_in_use = TRUE
```

File: code/game/objects/objs.dm [hunk 10]
```diff
@@ -42,42 +47,45 @@
...
+			src.interact(M)
```
Comment (DominikPanic) [inline code/game/objects/objs.dm]:
```suggestion
			interact(M)
```

**Pull request merged**